pages/dashboard.js

```diff
@@ -1,17 +1,29 @@
-import Head from 'next/head';
-import { Button, Flex, Text, Code, Icon } from '@chakra-ui/core';
+import useSWR from 'swr';
 
-import { useAuth } from '@/lib/auth';
+import DashboardShell from '@/components/DashboardShell';
 import EmptyState from '@/components/EmptyState';
+import SiteTableSkeleton from '@/components/SiteTableSkeleton';
+import SiteTable from '@/components/SiteTable';
+import fetcher from '@/utils/fetcher';
+import { useAuth } from '@/lib/auth';
 
 const Dashboard = () => {
   const auth = useAuth();
+  const { data } = useSWR('/api/sites', fetcher);
 
-  if (!auth.user) {
-    return 'Loading...';
+  if (!data) {
+    return (
+      <DashboardShell>
+        <SiteTableSkeleton />
+      </DashboardShell>
+    );
   }
 
-  return <EmptyState />;
+  return (
+    <DashboardShell>
+      {data.sites ? <SiteTable sites={data.sites} /> : <EmptyState />}
```

Comment:
Btw, I used this approach as an inspiration for my site, and here I had to do `data?.sites` to avoid an error that's thrown before the data is loaded. 

I'm wondering how it worked out without the `?` for you. 🤔

Reply:
At this point, my API has returned data. `data.sites` is either `[]` or `[{}]` 👍